gameFunctions.js

```diff
@@ -0,0 +1,288 @@
+    //                     setupBGChannel(roles, bgChannel.id);
+    //                 });
+    //             } else {
+    //                 setupBGChannel(roles, bgChannel.id);
+    //             }
+    //         } else {
+    //             // createChannel().then((newChannelId) => {
+    //             //     createGroup(roles.filter(x => x.role == 'bg'), newChannelId);
+    //             // })
+    //         }
+    //     }
+    // });
+}
+
+const getFormattedRoles = (players) => {
+    let roleDisplay = 'Player Roles: \n';
+    players.sort((a,b) => a.id - b.id);
+    players.forEach((player) => {
+        roleDisplay += `\t<@${player.id}>`;
+        if (player.role) {
+            roleDisplay += `\t(${player.role.toUpperCase()})`;
+        }
+        roleDisplay += `\n`;
+    });
+    return roleDisplay;
+}
+
+const unarchiveChannel = (channelId) => {
+    return new Promise((resolve, reject) => {
+        request.post({
+            url: 'https://slack.com/api/conversations.unarchive',
+            json: true,
+            headers: {
+                Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+            },
+            body: {
+                channel: channelId,
+            }
+        }, (error, response, rawBody) => {
+            console.log(rawBody);
+            if (error) {
+                console.log(error);
+                reject();
+            }
+            resolve();
+        });
+    });
+}
+
+const setupBGChannel = (players, channelId) => {
+    console.log('Removing players from BG Channel');
+    kickGroup(players, channelId).then(() => {
+        console.log('Adding players to bg channel');
+        createGroup(players.filter(x => x.role == 'bg'), channelId);
+    });
+}
+
+const notifyPlayers = (players) => {
+    players.forEach((player) => {
+        request.post({
+            url: 'https://slack.com/api/chat.postMessage',
+            json: true,
+            headers: {
+                Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+            },
+            body: {
+                channel: player.id,
+                text: getRoleText(player.role)
+            }
+        }, (error, response, rawBody) => {
+            if (error) {
+                console.log(error);
+            }
+        });
+    });
+
+}
+
+const kickGroup = (players, channelId) => {
+    let response = new Promise((resolve, reject) => {
+        let count = 0;
+        players.forEach(player => {
+            request.post({
+                url: 'https://slack.com/api/conversations.kick',
+                json: true,
+                headers: {
+                    Authorization: 'Bearer ' + process.env.SLACK_AUTH_TOKEN
+                },
+                body: {
+                    channel: channelId,
+                    user: player.id
+                }
+            }, (err, res, body) => {
+                console.log(body);
```

Comment:
Another console.log, Do we want to keep these?

Reply:
for now yes. Might remove them in refactoring